1.0/articles/routing.md

```diff
@@ -0,0 +1,165 @@
+
+## Introduction
+
+Polymer is designed with a set of principles in mind: **encapsulation**, **composition**, and **separation of concerns**. When we set out to tackle routing, we knew we wanted to do it in a way that adheres to these principles.
+
+Most client side routers are monolithic black boxes that require a complete and centralized route configuration with perfect knowledge of every routable location in the app. It’s also common for routing to be conflated with other concerns, like loading data or transitioning between views.
+
+In contrast, the carbon-route elements are a minimal, decoupled, and decentralized solution to routing in your app. With carbon-route, you layer routing into the reusable elements that make up your app.
+
+You can then use the data binding system to integrate with other elements like &lt;neon-animated-pages> for transitioning views, and &lt;iron-ajax> for loading data. And because these concerns are decoupled from routing, it’s easy to use other implementations, like &lt;iron-pages> if you don’t need animation, or &lt;firebase-element> to load data from a realtime database .
+
+## Example
+
+Ok, enough philosophy, let’s get into some examples.
+
+Here’s a simple UI with tabs and animated transitions:
+
+{%raw%}
+
+    <carbon-route pattern="/tabs/:tabName" data="{{data}}"></carbon-route>
+
+    <paper-tabs selected='{{data.tabName}}' attr-for-selected='key'>
+      <paper-tab key='foo'>Foo</paper-tab>
+      <paper-tab key='bar'>Bar</paper-tab>
+      <paper-tab key='baz'>Baz!</paper-tab>
+    </paper-tabs>
+
+    <neon-animated-pages selected='{{data.tabName}}'
+                         attr-for-selected='key'
+                         entry-animation='slide-from-left-animation'
+                         exit-animation='slide-right-animation'>
+      <neon-animatable key='foo'>Foo Page Here</neon-animatable>
+      <neon-animatable key='bar'>Bar Page Goes Here</neon-animatable>
+      <neon-animatable key='baz'>Baz Page, the Best One of the Three</neon-animatable>
+    </neon-animated-pages>
+
+Here the &lt;carbon-route> is data bound to both the tabs and the currently displayed page. So when the route changes, the selected tab and the currently displayed page automatically get updated too. And likewise, when the user selects a new tab, the route also updates in turn.
+
+Note also, how little actual routing there is here. Almost all of the work is being done by other elements that are purpose built for their roles in the UI, and they're just data bound to &lt;carbon-route>. This is right and proper.
+
+You can see the core of the idea here, but we’ve jumped ahead a bit. I mean, where does the route in `route="{{route}}"` come from? What’s that `pattern` property doing, and how is it related to `data`?
+
+{%endraw%}
+
+## &lt;carbon-route>
+
+Let’s walk back a bit and consider the consider the routing problem, one piece at a time, beginning with just a &lt;carbon-route>. What does it do, and how does it work?
```

Comment:
`Let's walk back a bit and consider the consider the` should be changed to 
`Let's walk back a bit and consider the`